drivers/flash/flash_nxp_s32_qspi.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023-2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_FLASH_NXP_S32_QSPI_H_
+#define ZEPHYR_DRIVERS_FLASH_NXP_S32_QSPI_H_
+
+#include "jesd216.h"
+
+#define QSPI_ERASE_VALUE      0xff
+#define QSPI_WRITE_BLOCK_SIZE 1U
+
+#define QSPI_IS_ALIGNED(addr, bits) (((addr) & BIT_MASK(bits)) == 0)
+
+#if defined(CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME)
+/* Size of LUT */
+#define QSPI_SFDP_LUT_SIZE     130U
+/* Size of init operations */
+#define QSPI_SFDP_INIT_OP_SIZE 8U
+#if defined(CONFIG_FLASH_JESD216_API)
+/* Size of all LUT sequences for JESD216 operations */
+#define QSPI_JESD216_SEQ_SIZE 8U
+#endif /* CONFIG_FLASH_JESD216_API */
+#endif /* CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME */
+
+struct nxp_s32_qspi_config {
+	const struct device *controller;
+	struct flash_parameters flash_parameters;
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
+	struct flash_pages_layout layout;
+#endif
+#if !defined(CONFIG_FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME)
```

Comment:
this currently only apply to NOR flash memories because of the HAL limitation you mentioned. If we want to keep it in the common driver sources, then drop the `_NOR_` and make it depend on `FLASH_NXP_S32_QSPI_NOR`. All SFDP runtime code can be moved to the common driver then.

Reply:
I updated